cmd/clusterctl/clusterdeployer/clusterclient/clusterclient.go

```diff
@@ -649,16 +657,83 @@ func (c *client) WaitForResourceStatuses() error {
 				klog.V(10).Info("retrying: machine status is empty")
 				return false, nil
 			}
-			// if m.Status.ProviderStatus == nil {
-			// 	klog.V(10).Info("retrying: machine.Status.ProviderStatus is not set")
-			// 	return false, nil
-			// }
 		}
 
 		return true, nil
 	})
 }
 
+func (c *client) GetClusterSecrets(cluster *clusterv1.Cluster) ([]*corev1.Secret, error) {
+	list := &corev1.SecretList{}
+	if err := c.clientSet.List(ctx, list, ctrlclient.InNamespace(cluster.Namespace)); err != nil {
+		return nil, errors.Wrapf(err, "error listing Secrets for Cluster %s/%s", cluster.Namespace, cluster.Name)
+	}
+
+	res := []*corev1.Secret{}
+	for _, secret := range list.Items {
+		if strings.HasPrefix(secret.Name, cluster.Name) {
+			res = append(res, &secret)
```

Comment:
This will always add the same address to the array - you need to do `secret := list.Items[i]` instead.